realm/realm-library/src/main/java/io/realm/internal/OsRealmConfig.java

```diff
@@ -232,12 +238,20 @@ private OsRealmConfig(final RealmConfiguration config,
         if (initializationCallback != null) {
             nativeSetInitializationCallback(nativePtr, initializationCallback);
         }
+
+        URI resolvedRealmURI  = null;
         // Set sync config
         if (syncRealmUrl != null) {
-            nativeCreateAndSetSyncConfig(nativePtr, syncRealmUrl, syncRealmAuthUrl, syncUserIdentifier,
-                    syncRefreshToken, sessionStopPolicy);
+            String resolvedSyncRealmUrl = nativeCreateAndSetSyncConfig(nativePtr, syncRealmUrl, syncRealmAuthUrl, syncUserIdentifier,
+                    syncRefreshToken, isPartial, sessionStopPolicy);
+            try {
+                resolvedRealmURI = new URI(resolvedSyncRealmUrl);
+            } catch (URISyntaxException e) {
+                RealmLog.error(e, "Cannot create a URI from the Realm URL address");
+            }
             nativeSetSyncConfigSslSettings(nativePtr, syncClientValidateSsl, syncSslTrustCertificatePath);
         }
+        this.resolvedRealmURI = resolvedRealmURI;
```

Comment:
Why is this needed? Doesn't the SyncConfiguration have the full URL?

Reply:
no, the ObjectStore will resolve the URL differently when using a partial sync https://github.com/realm/realm-object-store/pull/547/files#diff-864223d710d7b7e687e4c562de76f750R45